sklearn/model_selection/_search.py

```diff
@@ -555,6 +559,26 @@ def classes_(self):
         self._check_is_fitted("classes_")
         return self.best_estimator_.classes_
 
+    def _generate_candidates(self):
+        """Supplies candidates to search in response to accumulated results
+
+        To be overridden by implementers.
+
+        As in the following snippet, an implementation should yield an
+        initial list of candidates, where each candidate is a dict of
+        parameter settings.  The yield expression then returns the results
```

Comment:
is there a simple thing to add to this doc string to illustrate the "format" of the results dict? Or should we make a small example for the gallery that shows what to expect? I probably prefer the latter option and would try to make an example